wsd/DocumentBroker.cpp

```diff
@@ -2918,9 +2918,9 @@ std::size_t DocumentBroker::removeSession(const std::shared_ptr<ClientSession>&
                 // Don't destroy just yet, wait until save and upload are done.
                 // Notice that the save and/or upload could have been triggered
                 // earlier, and not necessarily here when removing this last session.
-                _docState.setUnloadRequested();
```

Comment:
This is the only place this is set; but it is checked in a lot of places - we should remove the state if we're going down this path; but then we need to check that all the 'markToDestroy' code-paths are instead gated on something else: eg. "no open sessions" at those points instead (?) =) or I fear we won't destroy the document (?)

Anyhow - looks encouragingly simpler than expected which is good.